notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+    "      EXTERN(\n",
+    "        '{\"type\":\"http\",\n",
+    "          \"uris\":[ \"https://druid.apache.org/data/wikipedia.json.gz\"]\n",
+    "         }',\n",
+    "        '{\"type\":\"json\"}'\n",
+    "      )\n",
+    ") EXTEND (\"isRobot\" VARCHAR, \"channel\" VARCHAR, \"timestamp\" VARCHAR, \"flags\" VARCHAR, \"isUnpatrolled\" VARCHAR, \"page\" VARCHAR, \"diffUrl\" VARCHAR, \"added\" BIGINT, \"comment\" VARCHAR, \"commentLength\" BIGINT, \"isNew\" VARCHAR, \"isMinor\" VARCHAR, \"delta\" BIGINT, \"isAnonymous\" VARCHAR, \"user\" VARCHAR, \"deltaBucket\" BIGINT, \"deleted\" BIGINT, \"namespace\" VARCHAR, \"cityName\" VARCHAR, \"countryName\" VARCHAR, \"regionIsoCode\" VARCHAR, \"metroCode\" BIGINT, \"countryIsoCode\" VARCHAR, \"regionName\" VARCHAR))\n",
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  *\n",
+    "FROM \"ext\"\n",
+    "\n",
+    "WHERE \"isRobot\"='false'\n",
+    "\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "druid.display.run_task(sql)\n",
+    "druid.sql.wait_until_ready('wikipedia_events_only_human')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c825d39-0bbd-4cd3-ae8a-760146d9fff7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "druid.display.sql(\"\"\"\n",
+    "SELECT isRobot, channel, count(*) num_events\n",
+    "FROM \"wikipedia_events_only_human\" \n",
+    "GROUP BY 1,2 \n",
+    "ORDER BY 3 DESC \n",
+    "LIMIT 10\n",
+    "\"\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6561d663-9cfe-4a62-80cc-dcbb0ce199ff",
+   "metadata": {},
+   "source": [
+    "#### Transform Data During Ingestion\n",
+    "\n",
+    "The SQL language provides a rich [set of functions](https://druid.apache.org/docs/latest/querying/sql-scalar.html) that can be applied to input columns to transform the data as it is being ingested. All scalar SQL function are available for normal ingestion. Rollup ingestion is discussed in the [Rollup Notebook](05-rollup.ipynb) which includes the use of aggregate functions at ingestion time as well.\n",
+    "\n",
+    "Here are some examples of such transformations:\n",
+    "\n",
+    "##### Time manipulation\n",
+    "There are many [time parsing and manipulation functions](https://druid.apache.org/docs/latest/querying/sql-scalar.html#date-and-time-functions) available in Apache Druid. It is common to do some time cleansing/transformation at ingestion. Here are some examples of time manipulation functions:\n",
+    "```\n",
+    "  TIME_PARSE( \"timestamp\") AS \"__time\",   \n",
+    "  TIME_FLOOR( TIME_PARSE( \"timestamp\"), 'P1W') AS \"week_start\",\n",
+    "  TIMESTAMPDIFF( DAY,\n",
+    "                 TIME_FLOOR( TIME_PARSE( \"timestamp\"), 'P1W'),\n",
+    "                 TIME_PARSE( \"timestamp\")\n",
+    "               ) AS \"days_since_week_start\"\n",
+    "   \n",
+    "```\n",
+    "\n",
+    "##### Use CASE statements to transform data\n",
```

Comment:
This section I intend to cover in a notebook on IF / CASE - when done we can add a link out to that notebook explicitly.

Reply:
... part of the problem of the PRs being all split up right now is that I don't see the other notebooks in this branch.... perhaps we need to do a pass at all notebooks to provide more cross notebook references in the learn more sections of all notebooks. But that is better done once we merge most PRs.